src/modlunky2/mem/arena_state.py

```diff
@@ -0,0 +1,356 @@
+    TIDEPOOL_2 = enum.auto()
+    TIDEPOOL_3 = enum.auto()
+    TIDEPOOL_4 = enum.auto()
+    TIDEPOOL_5 = enum.auto()
+    TEMPLE_1 = enum.auto()
+    TEMPLE_2 = enum.auto()
+    TEMPLE_3 = enum.auto()
+    TEMPLE_4 = enum.auto()
+    TEMPLE_5 = enum.auto()
+    ICECAVES_1 = enum.auto()
+    ICECAVES_2 = enum.auto()
+    ICECAVES_3 = enum.auto()
+    ICECAVES_4 = enum.auto()
+    ICECAVES_5 = enum.auto()
+    NEOBABYLON_1 = enum.auto()
+    NEOBABYLON_2 = enum.auto()
+    NEOBABYLON_3 = enum.auto()
+    NEOBABYLON_4 = enum.auto()
+    NEOBABYLON_5 = enum.auto()
+    SUNKENCITY_1 = enum.auto()
+    SUNKENCITY_2 = enum.auto()
+    SUNKENCITY_3 = enum.auto()
+    SUNKENCITY_4 = enum.auto()
+    SUNKENCITY_5 = enum.auto()
+
+
+ARENA_STATE_BASE = 0x95C
+# Used for both Arena Config and Arena Levels which both happen to be
+# 40-tuple of bools.
+TupleBool40 = Tuple[
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+    bool,
+]
+
+
+@serialize(rename_all="spinalcase")
+@deserialize(rename_all="spinalcase")
+@dataclass(frozen=True)
+class ArenaState:
+    format: ArenaFormat = struct_field(
+        0x964 - ARENA_STATE_BASE, sc_int8, default=ArenaFormat.DEATHMATCH
+    )
+    ruleset: ArenaRuleset = struct_field(
+        0x965 - ARENA_STATE_BASE, sc_int8, default=ArenaRuleset.CUSTOM
+    )
+    timer: ArenaTimer = struct_field(
+        0x978 - ARENA_STATE_BASE, sc_int8, default=ArenaTimer.TIME_02_00
+    )
+    timer_ending: ArenaTimerEnding = struct_field(
+        0x979 - ARENA_STATE_BASE, sc_int8, default=ArenaTimerEnding.ROUND_ENDS
+    )
+    wins: int = struct_field(0x97A - ARENA_STATE_BASE, sc_uint8, default=5)
+    lives: int = struct_field(0x97B - ARENA_STATE_BASE, sc_uint8, default=2)
+    time_to_win: ArenaTimeToWin = struct_field(
+        0x97C - ARENA_STATE_BASE, sc_int8, default=ArenaTimeToWin.SECONDS_30
+    )
+    health: int = struct_field(0x986 - ARENA_STATE_BASE, sc_uint8, default=4)
+    bombs: int = struct_field(0x987 - ARENA_STATE_BASE, sc_uint8, default=4)
+    ropes: int = struct_field(0x988 - ARENA_STATE_BASE, sc_uint8, default=4)
+    stun_time: ArenaStunTime = struct_field(
+        0x989 - ARENA_STATE_BASE, sc_int8, default=ArenaStunTime.X0_50
+    )
+    mount: ArenaMount = struct_field(
+        0x98A - ARENA_STATE_BASE, sc_int8, default=ArenaMount.NONE
+    )
+    arena_select: ArenaSelect = struct_field(
+        0x98B - ARENA_STATE_BASE, sc_int8, default=ArenaSelect.RANDOM_LEVEL
+    )
+    arenas: TupleBool40 = struct_field(
+        0x98C - ARENA_STATE_BASE,
+        array(sc_bool, 40),
+        default_factory=lambda: tuple([False] * 40),
```

Comment:
`default=(False,)*40` works too.  I don't have a strong preference

Reply:
ah yea i originally had a list before realizing the intent that they are immutable. I might keep it as a factory though just so the giant tuple is deferred